test/integration/targets/ansible-galaxy-collection/tasks/upgrade.yml

```diff
@@ -0,0 +1,166 @@
+##### Updating collections with a new version constraint
+
+# No deps
+
+- name: reset installation directory
+  file:
+    state: "{{ item }}"
+    path: "{{ galaxy_dir }}/ansible_collections"
+  loop:
+    - absent
+    - directory
+
+- name: install a collection
+  command: ansible-galaxy collection install namespace1.name1:0.0.1 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+  failed_when:
+    - '"namespace1.name1:0.0.1 was installed successfully" not in result.stdout_lines'
+
+- name: install a new version of the collection without --force
+  command: ansible-galaxy collection install namespace1.name1:>0.0.4,<=0.0.5 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: '"namespace1.name1:0.0.5 was installed successfully" in result.stdout_lines'
```

Comment:
Do you think we need a test to check the version in the MANIFEST.json. I'm somewhat concerned about relying on just the display output and would be nice to check the actual collection file for this and the subsequent tests. At least maybe for some of these tests it would be a good idea and then we can just rely on the output.

Reply:
Okay, I'll add that.